src/sentry/tasks/statistical_detectors.py

```diff
@@ -241,6 +303,31 @@ def _detect_transaction_trends(
     )
 
 
+def query_transactions_timeseries(
+    transactions: List[Tuple[int, int | str]],
+    start: datetime,
+    agg_function: str,
+) -> Generator[Tuple[int, int, Any], None, None]:
+    end = start.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+    interval = 3600  # 1 hour
+
+    for project_id, transaction_name in transactions:
+        params: Dict[str, Any] = {
+            "start": end - timedelta(days=14),
+            "end": end,
+            "project_id": [project_id],
+            "project_objects": [Project.objects.get(id=project_id)],
+        }
+        results = metrics_performance.timeseries_query(
+            selected_columns=[agg_function],
+            query=f"transaction:#{transaction_name}",
```

Comment:
what is the `#` for? Also, I think you have to check if the transaction name needs to be escaped.

Reply:
Thanks! # is old Ruby muscle memory. Do you know if we have a function for escaping query args already kicking around? 🤔

Reply:
Couldn't find a conclusive answer in [the docs](https://docs.sentry.io/product/sentry-basics/search/) but I'm assuming that `"` is the only char needing escaping if you're in quotes. If so, should be escaped now.